### ISSUE TITLE
Text out of bounds in Search results feed 

### ISSUE THREAD

**Issue by madarsbiss** [thepracticaldev/dev.to#8238]
**Describe the bug**

Today I was using DEV search and noticed text out of bounds in one of the results.

**To Reproduce**

Search: https://dev.to/search?q=project%20ideas
Scroll down till article below shows up (should be around 20th-25th result).

**Screenshots**
![Screenshot](https://user-images.githubusercontent.com/42511917/83550981-5efe3800-a510-11ea-98c2-b33f0fd6c4bb.png)

**Desktop (please complete the following information):**

- OS: Win10 x64
- Browser: Chrome
- Version: 83.0.4103.61

I did check on Firefox too (v 77.0), the problem still exists.

**Smartphone (please complete the following information):**

- Device:
- OS:
- Browser:
- Version:

**Additional context**

**Comment by Ridhwana** [comment]
Thanks for the issue! We'll take your request into consideration and follow up if we decide to tackle this issue.

To our amazing contributors: issues labeled `type: bug` are always up for grabs, but for feature requests, please wait until we add a `ready for dev` before starting to work on it.

To claim an issue to work on, please leave a comment. If you've claimed the issue and need help, please ping @thepracticaldev/oss and we will follow up within 3 business days.

For full info on how to contribute, please check out our [contributors guide](https://github.com/thepracticaldev/dev.to/blob/master/CONTRIBUTING.md).

**Comment by icncsx** [comment]
Looks like the issue is the non-breaking space: `&nbsp`. 

![Screenshot from 2020-06-03 12-53-40](https://user-images.githubusercontent.com/60417145/83678999-a3d7ac80-a5ac-11ea-97d4-4d9b55b81324.png)

@Ridhwana, it looks like the overwhelming majority of articles don't have that issue. Do we want to handle this server side or client side?

If we want to change it client side, it's a pretty trivial fix on line 221 [here](https://github.com/thepracticaldev/dev.to/blob/master/app/assets/javascripts/utilities/buildArticleHTML.js#L221). 

```js
bodyTextSnippet =
        startingEllipsis + article.highlight.body_text.map(x=>x..replace(/&nbsp;/g, " ")).join('...') + '…';
```

If we want to change it server side, it really depends on how downstream or upstream we want to 
gsub out the `&nbsp`. Upon cursory inspection, a good candidate seems to be the `prepare_doc` method in [feed_content.rb](https://github.com/thepracticaldev/dev.to/blob/master/app/services/search/feed_content.rb#L22) where we make sure that `hit["highlight"]["body_text"]` is sanitized.

Note that we also have the same issue for the full body text as well because snippet and full body use the same data. So the issue with the full body text can either be another issue or the scope of this issue can be expanded to be more inclusive.

![nbsp](https://user-images.githubusercontent.com/60417145/83696666-0770d280-a5cb-11ea-94f4-b1129a344631.png)

Let me know your thoughts :)

**Comment by madarsbiss** [comment]
@icncsx @Ridhwana as I see this, all the content from text editor should always be processed (properly formatted) at post stage.. That would store the data properly and ultimately avoid cases like these in future..

**Comment by icncsx** [comment]
@madarsbiss Totally agree. That can either be a separate linked issue or the scope of this issue and accompanying PR can expand and address both needs: the need to format already "unpure" articles and the need to process article title and body text upon posting and storage into DB.
